test/integration/eslint/test/index.test.js

```diff
@@ -97,10 +97,13 @@ describe('ESLint', () => {
       const eslintrc = join(dirFirstTimeSetup, '.eslintrc')
       await writeFile(eslintrc, '')
 
-      const { stdout, stderr } = await nextLint(dirFirstTimeSetup, [], {
-        stdout: true,
-        stderr: true,
-      })
+      const { stdout, stderr } = await nextLint(
+        ['--base-dir', dirFirstTimeSetup],
```

Comment:
This looks like a breaking change since `--base-dir` was not necessary before.

Do we need to bump semver major?

Reply:
`--base-dir` isn't a required field and defaults to the current working directory, but yes it is breaking for those who were specifying it before without the flag.

I also removed `--dir` as a flag from `next lint`. File, or directory, paths should be passed in without any flags similar to `eslint` so this was redundant (and I made this change to fix #26564)

>  Do we need to bump semver major?

Yes we should. I'm okay with holding off on merging this until we bump semver major.

I also added a [section](https://github.com/vercel/next.js/pull/26777/files#diff-0220ca71b6e5d1c80ad9e3dedd642059ed9a26c2a5ae414e46f022d2e688c9a6) to the Upgrading doc.

Reply:
@timneutkens Are we planning to do a 12 release soon? We should probably wait to merge this PR if not.

Reply:
Maybe we could allow testing the new arg handling behind an experimental flag in `next.config.js` to allow getting feedback before changing it in a major

Reply:
In `next lint`, we would need to know which directory Next.js is in before we can load the user's `next.config.js` right? Unfortunately, specifying the location of the Next.js directory is what's being changed here (`--base-dir`), so that might make things a little tricky 🤔